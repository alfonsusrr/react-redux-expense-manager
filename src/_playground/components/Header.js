import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => {
    const activeClassName = 'is-active'
    return (
        <div>
            <h1>Portfolio</h1>
            <nav>
                <NavLink to='/' className={({isActive}) => isActive ? activeClassName : undefined}>Home</NavLink>
                <NavLink to='/portfolio' className={({isActive}) => isActive ? activeClassName : undefined}>Portfolio</NavLink>
                <NavLink to='/contact' className={({isActive}) => isActive ? activeClassName : undefined}>Contact</NavLink>
            </nav>
        </div>
    )
}

export default Header