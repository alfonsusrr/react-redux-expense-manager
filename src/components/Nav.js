import React from 'react'
import { NavLink } from 'react-router-dom'

const Nav = () => {
    const activeClassName = 'is-active'

    return (
        <nav>
            <NavLink to="/" 
                     className={({isActive}) => isActive ? activeClassName : undefined }
            >
                Home
            </NavLink>
            <NavLink to="/create" 
                     className={({isActive}) => isActive ? activeClassName : undefined }
            >
                Add Expense
            </NavLink>
            <NavLink to="/edit" 
                     className={({isActive}) => isActive ? activeClassName : undefined }
            >
                Edit Expense
            </NavLink>
            <NavLink to="/help" 
                     className={({isActive}) => isActive ? activeClassName : undefined }
            >
                Help
            </NavLink>
        </nav>
    )
}

export default Nav