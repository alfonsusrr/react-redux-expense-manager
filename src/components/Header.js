import React from 'react'
import Nav from './Nav'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <Nav/>
    </header>
)

export default Header