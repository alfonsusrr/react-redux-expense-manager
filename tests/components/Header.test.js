import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import Header from '../../src/components/Header'
import Nav from '../../src/components/Nav'

test('Should render Header correctly', () => {
    const renderer = new ReactShallowRenderer()
    renderer.render(<Header/>)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
})

test('Should render Nav correctly', () => {
    const renderer = new ReactShallowRenderer()
    renderer.render(<Nav/>)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
})
