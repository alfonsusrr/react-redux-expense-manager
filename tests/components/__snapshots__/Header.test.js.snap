// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <Nav />
</header>
`;

exports[`Should render Nav correctly 1`] = `
<nav>
  <NavLink
    className={[Function]}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    className={[Function]}
    to="/create"
  >
    Add Expense
  </NavLink>
  <NavLink
    className={[Function]}
    to="/edit"
  >
    Edit Expense
  </NavLink>
  <NavLink
    className={[Function]}
    to="/help"
  >
    Help
  </NavLink>
</nav>
`;
